command/agent/config.go

```diff
@@ -849,6 +854,9 @@ func MergeConfig(a, b *Config) *Config {
 	if b.BindAddr != "" {
 		result.BindAddr = b.BindAddr
 	}
+	if b.HttpPort != 0 {
+		result.Ports.HTTP = b.HttpPort
+	}
```

Comment:
Can we add a test case to ensure this merging works? There are examples in `config_test.go` that you can piggyback on.